core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -202,6 +214,156 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
       WHERE workspace_id = $workspaceId
       GROUP BY entity_type;""".as[EntityTypeAndCount]
 
+  def countEntities(workspaceId: UUID, entityType: String): ReadWriteAction[Int] =
+    sql"""select count(*) #$fromEntityWhereNotDeleted
+              and e.entity_type = $entityType
+              and e.workspace_id = $workspaceId
+       """
+      .as[Int]
+      .map(_.head)
+
+  def countEntitiesWithColumnFilter(workspaceId: UUID,
+                                    entityType: String,
+                                    columnFilter: EntityColumnFilter
+  ): ReadWriteAction[Int] =
+    // CAST, JSON_UNQUOTE and JSON_EXTRACT are used to handle strings and numbers and do a case insensitive comparison
+    sql"""select count(*) #$fromEntityWhereNotDeleted
+              and e.entity_type = $entityType
+              and e.workspace_id = $workspaceId
+              and CAST(JSON_UNQUOTE(JSON_EXTRACT(e.attributes,
+                ${slickAttributePath(columnFilter.attributeName)})) AS CHAR) = ${columnFilter.term}
+       """
+      .as[Int]
+      .map(_.head)
+
+  def countEntitiesWithFilterTerms(workspaceId: UUID,
+                                   entityType: String,
+                                   entityQuery: EntityQuery
+  ): ReadWriteAction[Int] =
+    concatSqlActions(
+      sql"""select count(*)
+            #$fromEntityWhereNotDeleted
+            and e.entity_type = $entityType
+            and e.workspace_id = $workspaceId
+       """,
+      buildFilterConditions(entityQuery)
+    )
+      .as[Int]
+      .map(_.head)
+
+  private def buildFilterConditions(entityQuery: EntityQuery) = {
+    val filterClauses = entityQuery.filterTermsList.map { filterTerm =>
+      sql"""JSON_SEARCH(e.attributes, 'one', ${'%' + filterTerm.toLowerCase + '%'})"""
+    }
```

Comment:
the current functionality in LocalEntityProvider is exact-match for column filters, not a substring match. I think it would be great to enable substring matching but if there's any performance impact or other issue, we'd be ok with leaving it as exact match.